python3/dnf_plugins/xapitoken.py

```diff
@@ -0,0 +1,45 @@
+"""dnf plugin to set xapitoken http header for enabled repos"""
+import json
+import logging
+# Disable the error, it can be import in production env
+# and mocked out in unitttest
+# pylint: disable=import-error
+# pytype: disable=import-error
+import dnf
+import urlgrabber
+
+
+class InvalidToken(Exception):
+    """Token is invlaid"""
+    def __init__(self, token):
+        super().__init__(f"Invalid token: {token}")
+
+
+#pylint: disable=too-few-public-methods
+class AccessToken(dnf.Plugin):
+    """dnf xapitoken plugin class"""
```

Comment:
"xapi token"

Reply:
Updated `class AccessToken` to `class XapiToken`, and I think the "xapitoken" here is the name of the plugin